Docs 4556 for each scope

Cristian-Venticinque opened the pull request.

Commit:
Styling updates. 
-Added *bold* and `backticks` where necessary.
-Removed unnecessary details in parenthesis.

Commit:
Clarified where Mule stores the items in the collection.

Commit:
Added an example XML aligned with the Flow showed as an example.

Commit:
Changed title for disambiguation

Commit:
Merged XML reference content into this article.

Commit:
Removed unnecessary See Also link.

Commit:
Deleted file. Content was moved to for-each-scope-concept.adoc

elrodro83 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -16,24 +16,37 @@ The For Each scope performs a similar task to using a Splitter and then an Aggre
 image::component-foreach-example.png[For Each Component]
 
-
+Each item in the collection is stored in the payload when processed through the For Each Scope. For example, in the scenario detailed above each split item is stored in `payload.topic` during each iteration.
```
this should be just `payload`, not `payload.topic`. Can you check in the example?

Cristian-Venticinque replied:
You are correct, I tested locally and the split items are stored in `payload`. I'll deleted the second sentence here and will update the first one.

Commit:
Addressed change requested during review. 

-The example was incorrect, clarified this behavior. 
-Fixed minor capitalization errors.

dhirsch1 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -46,6 +59,34 @@ There are several example projects in Anypoint Exchange that you can open in Any
+|Property | Default | Description
+| `collection`
+| `payload`
+| An expression to that returns a Java collection, object array, map, or DOM
```
"An expression that returns..."

dhirsch1 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -16,24 +16,37 @@ The For Each scope performs a similar task to using a Splitter and then an Aggre
 
 * By default For Each tries to split the payload. If the payload is a simple Java collection, the For Each scope can split it without any configuration.
 * For Each will not modify the current payload.
-* For non-Java collections, such as XML or JSON, you need to use a DateWeave expression to split data. You use the Collection (`collection`) field for this purpose.
```
"Use the Collection..." The word "You" is not needed.

dhirsch1 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -16,24 +16,37 @@ The For Each scope performs a similar task to using a Splitter and then an Aggre
 * By default For Each tries to split the payload. If the payload is a simple Java collection, the For Each scope can split it without any configuration.
 * For Each will not modify the current payload.
-* For non-Java collections, such as XML or JSON, you need to use a DateWeave expression to split data. You use the Collection (`collection`) field for this purpose.
+* For non-Java collections, such as XML or JSON, you need to use a DateWeave expression to split data. You use the *Collection* field for this purpose.
```
Same comment - "Use the ..."

dhirsch1 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -16,24 +16,37 @@ The For Each scope performs a similar task to using a Splitter and then an Aggre
+* For non-Java collections, such as XML or JSON, you need to use a DateWeave expression to split data. You use the *Collection* field for this purpose.
 +
-Here, the Collection field in For Each is set to iterate over an array stored in `payload.topics`.
+Here, the *Collection* field in For Each is set to iterate over an array stored in `payload.topics`:
```
Instead of "Here," I would say something like this: "The following example"

dhirsch1 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -16,24 +16,37 @@ The For Each scope performs a similar task to using a Splitter and then an Aggre
 ////
 
-You can also split an array in batches. Potentially, batches produce quicker processing. Each batch is treated as a separate Mule message. For example, if a collection has 200 elements and you set Batch Size (`batchSize`) to `50`, the For Each scope iteratively processes 4 batches of 50 elements, each as a separate Mule message.
+You can also split an array in batches. Potentially, batches produce quicker processing. Each batch is treated as a separate Mule message. For example, if a collection has 200 elements and you set *Batch Size* to `50`, the For Each scope iteratively processes 4 batches of 50 elements, each as a separate Mule message.
```
"You can also split an array into batches to enable quicker processing."

dhirsch1 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -16,24 +16,37 @@ The For Each scope performs a similar task to using a Splitter and then an Aggre
 If one of the elements in a collection throws an exception, the For Each scope stops processing that collection and invokes the error handler.
 
-== Examples
+== Example projects
```
Capitalize "projects" for consistency.

dhirsch1 commented on modules/ROOT/pages/for-each-scope-concept.adoc:
```diff
@@ -3,7 +3,7 @@ ifndef::env-site,env-github[]
 endif::[]
 
-The For Each scope (`foreach`) splits a payload into elements and processes them one by one through the components that you place in the scope. It is similar to a for-each/for-loop code block in most programming languages.
+The For Each scope splits a payload into elements and processes them one by one through the components that you place in the scope. It is similar to a `for-each`/`for` loop code block in most programming languages.
```
I suggest merging the next paragraph with this one: 

...It is similar to a `for-each`/`for` loop code block in most programming languages and can process any collection, including lists and arrays. The collection can be any supported content type, such as `application/json`, `application/java`, or `application/xml`.

The bulleted below the paragraph list should have an introduction: "The For Each scope works as follows:"

Commit:
Addressed changes requested during review

-Fixed spelling and grammar issues.
-Improved wording in some paragraphs.
-Minor styling updates

Commit:
Minor wording update.
